tests/test_model.py

```diff
@@ -0,0 +1,84 @@
+                        job_id="0001",
+                        criteria=criteria)
+
+    assert result == expected
+
+def test_save_load():
+    c = [model.Criterion(name="PhD",
+                         description="Degree or relevant experience",
+                         scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")),
+         model.Criterion(name="Research software",
+                         description="Authorship,development and maintenance",
+                         scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")),
+         model.Criterion(name="Best practices",
+                         description="Issue tracking, testing, documentation etc.",
+                         scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent"))]
+    
+    a = [model.Applicant(name="George Smith",
+                         cv="placeholder",
+                         scores={model.Criterion(name="PhD",
+                                                 description="Degree or relevant experience",
+                                                 scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")): "Excellent",
+                                 model.Criterion(name="Research software",
+                                                 description="Authorship,development and maintenance",
+                                                 scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")): "Satisfactory"}),
+         model.Applicant(name="Jim Chapman",cv="placeholder",scores={model.Criterion(name="PhD",
+                                                                                     description="Degree or relevant experience",
+                                                                                     scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")): "Moderate",
+                                                                     model.Criterion(name="Research software",
+                                                                                     description="Authorship,development and maintenance",
+                                                                                     scores = ("Unsatisfactory","Moderate","Satisfactory","Excellent")): "Unsatisfactory"})]
+
+    shortlist = model.Shortlist(role= model.Role(job_title="tests",
+                                                job_id="0000",
+                                                criteria=c),
+                               applicants=a)
+    
+    model.save_shortlist(Path("tests"), shortlist)
+    result:model.Shortlist = model.load_pickle(Path("tests")/pickle_file_name)
+
+    assert result.role.job_title == "tests"
+    assert result.role.job_id == "0000"
+    assert result.role.criteria == c
+    assert result.applicants == a 
+    assert result.applicants[0].name == "George Smith"
+    assert result.applicants[1].name == "Jim Chapman"
+
```

Comment:
Add a check that the applicant's score's Criterion keys are in the Role's list of criterion.